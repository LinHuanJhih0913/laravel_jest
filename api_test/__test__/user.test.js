/**
 * @jest-environment node
 */

const axios = require('axios');
const baseURI = 'http://127.0.0.1:8000';
const {matchers} = require('jest-json-schema');

expect.extend(matchers);

it('fetch api/users/1 schema', async () => {
    const schema = {
        'type': 'object',
        'required': [
            'id', 'name', 'email', 'email_verified_at', 'created_at', 'updated_at'
        ],
        'properties': {
            'id': {'type': 'integer',},
            'name': {'type': 'string',},
            'email': {'type': 'string',},
            'email_verified_at': {'type': 'string',},
            'created_at': {'type': 'string',},
            'updated_at': {'type': 'string',}
        }
    };
    expect((await fetchUsersByID(1)).data).toMatchSchema(schema);
});

test('fetch api/users/1 snapshot', async () => {
    expect((await fetchUsersByID(1)).data).toMatchSnapshot();
});

it('fetch api/users schema', async () => {
    const schema = {
        'type': 'array',
        'items': {
            'type': 'object',
            'required': [
                'id', 'name', 'email', 'email_verified_at', 'created_at', 'updated_at'
            ],
            'properties': {
                'id': {'type': 'integer',},
                'name': {'type': 'string',},
                'email': {'type': 'string',},
                'email_verified_at': {'type': 'string',},
                'created_at': {'type': 'string',},
                'updated_at': {'type': 'string',}
            }
        }
    };
    expect((await fetchUsers()).data).toMatchSchema(schema);
});

test('fetch api/users snapshot', async () => {
    expect((await fetchUsers()).data).toMatchSnapshot();
});

function fetchUsersByID(id) {
    return axios.get(`${baseURI}/api/users/${id}`);
}

function fetchUsers() {
    return axios.get(`${baseURI}/api/users`);
}
