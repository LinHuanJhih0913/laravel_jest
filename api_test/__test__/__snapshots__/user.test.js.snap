// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch api/users snapshot 1`] = `
Array [
  Object {
    "created_at": "2019-05-18 09:19:28",
    "email": "jacinto25@example.org",
    "email_verified_at": "2019-05-18 09:19:28",
    "id": 1,
    "name": "Minnie Wiza",
    "updated_at": "2019-05-18 09:19:28",
  },
  Object {
    "created_at": "2019-05-18 09:19:37",
    "email": "hortense56@example.net",
    "email_verified_at": "2019-05-18 09:19:37",
    "id": 2,
    "name": "Mrs. Corene Bauch DVM",
    "updated_at": "2019-05-18 09:19:37",
  },
  Object {
    "created_at": "2019-05-18 09:19:40",
    "email": "nelda.boyer@example.com",
    "email_verified_at": "2019-05-18 09:19:40",
    "id": 3,
    "name": "Mr. Karl Bradtke",
    "updated_at": "2019-05-18 09:19:40",
  },
]
`;

exports[`fetch api/users/1 snapshot 1`] = `
Object {
  "created_at": "2019-05-18 09:19:28",
  "email": "jacinto25@example.org",
  "email_verified_at": "2019-05-18 09:19:28",
  "id": 1,
  "name": "Minnie Wiza",
  "updated_at": "2019-05-18 09:19:28",
}
`;
